### ISSUE TITLE
LCD Does not work upon flash

### ISSUE THREAD

**Issue by CampbellFabrications** [MalyanSystem/Malyan-M180-Marlin#3]
I purchased a Malyan M180 from Jaycar Electronics under the name Duinotech TL4090
I have successfully flashed Marlin to the board via the following command
`avrdude -C"/Applications/Arduino.app/Contents/Java/hardware/tools/avr/etc/avrdude.conf" -v -pm2560 -cstk500v1 -P/dev/tty.wch1420 -b57600 -D -Uflash:w:/private/var/folders/zp/j0bc64jx59g83yq0h3fvn_j40000gn/T/build6569504525374575330.tmp/Marlin.cpp.hex`

The Uflash:w: following code will vary between user.
tty.wch1420 is the serial name for the board recognised by my Mac.
CSRutil has been disabled for SU access.

When starting the printer up via switch, the LCD will switch on briefly, then turn off.
If I disconnect the LCD while the printer is on, and reconnect, it shows the Duinotech text on a white screen (Presumably an EEPROM logo since I've previously executed all of the wiki.malyansys Sailfish exe files to update the M180).

I can connect to the printer via USB in Simplify3D and control it via Pronterface, however the LCD panel and encoder are not working.
the SD card can be read by Simplify3D as well as M20 and M21 returning successful results.

Photos of the disassembled printer included for posterity http://imgur.com/a/qr0dy

**Comment by yongzong** [comment]
Hi,
1. You can contact support@malyansys.com directly.
2. If you flash back to sailfish firmware and still not working, you need
   to double check the connection between motherboard and LCD. The
   communication is simple UART protocol, you can use use multi-meter to
   measure connection or oscillator to check signal level.

BR,
Yongzong Lin

2016-08-24 5:56 GMT+08:00 SydneyMakers notifications@github.com:

**Comment by leonardojc** [comment]
i've the same problem!
i dont know how to solve it
with the original FW, the LCD work Fine
Any ideas?

**Comment by CampbellFabrications** [comment]
Sorry for not replying to this issue, I have successfully flashed Marlin to the board and have the LCD working, albeit it is on a firmware i no longer possess however i do still recall where the modification is needed and how it affects the original LCD.

So the M180 initialises a Serial connection with the LCD in LcdBoard.cpp function `malyan_lcd_init()`

There are 3 hardcoded `delay(1000)` calls between each command being sent to the LCD, and from the looks of it, it is a failsafe designed to attempt multiple connections and going with whatever connects.
after extending the third `delay(1000)` to `delay(3000)` (Line 105) seems to do the trick, allowing the hardware to power up successfully before attempting the serial connection.

**Comment by CampbellFabrications** [comment]
Sorry for not replying to this sooner, The issue with the LCD stems from the LcdBoard.cpp file and the `malyan_lcd_init()` function.

In short, the LCD isnt powered on by the time the main board sends the connection request. This is done on line 101:
` writeString((char *)"{SYS:STARTED}");`

By adding a `delay(1000);` before this, we ensure the screen has enough time to boot before the main board polls the LCD.

You may need to increase this value, unfortunately it has varied between 4 M180 printers i've tested this on.

**Comment by specky532** [comment]
Hi campbell,

Is all you did the delay? I've tried adding this in and tried values ranging from 1000 all the way to 10000 and I still can't get the LCD to turn on after the initial duinotech splashscreen. I have the exact same printer (the duinotech one) as you.

Cheers,
Ben

> Sorry for not replying to this sooner, The issue with the LCD stems from the LcdBoard.cpp file and the `malyan_lcd_init()` function.
> 
> In short, the LCD isnt powered on by the time the main board sends the connection request. This is done on line 101:
> ` writeString((char *)"{SYS:STARTED}");`
> 
> By adding a `delay(1000);` before this, we ensure the screen has enough time to boot before the main board polls the LCD.
> 
> You may need to increase this value, unfortunately it has varied between 4 M180 printers i've tested this on.

**Comment by CampbellFabrications** [comment]
All I did was add the delay after assuming that since it existed but was
originally commented out implies there was firmware testing underway when
the printer was released and never got around to fixing them

**Comment by specky532** [comment]
Yeah, I managed to find a working copy on the facebook group (Pretty sure it was actually you who uploaded it). I have the LED working now but can't seem to get the left extruder fan or extruder at all to turn on or work. Did you ever manage to get both extruders working on yours with marlin?

**Comment by CampbellFabrications** [comment]
I never really progressed that far after the whole LCD issue, and yes, that
was my upload.
The perks of working for a place that sells said printer; there were always
PER's to play with
